### PR TITLE
rgw: fix indentation for listobjectsv2

### PR DESCRIPTION
**Pull request by albin-antony**
Signed-off-by: Albin Antony <aantony@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by albIN7** [comment]
@alimaredia Please review

**Comment by alimaredia** [comment]
@albIN7 There are more things you missed in: https://github.com/ceph/ceph/pull/28102/files. 

For one, the body of RGWListBucket_ObjStore_S3::get_common_params() is not indented at all. Could you please look through the entire PR one more time and make sure everything follows these guidelines:

https://github.com/ceph/ceph/blob/master/CodingStyle

Any other coding style changes can go into this PR.

**Comment by albIN7** [comment]
My bad! Is this fine?

**Comment by alimaredia** [comment]
@albIN7 can you fix the merge conflict here?

**Comment by albIN7** [comment]
> @albIN7 can you fix the merge conflict here?

Done

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
 int RGWListBucket_ObjStore_S3::get_common_params()
-
 {
-list_versions = s->info.args.exists("versions");
-prefix = s->info.args.get("prefix");
+  list_versions = s->info.args.exists("versions");
+  prefix = s->info.args.get("prefix");
 
 // non-standard
-s->info.args.get_bool("allow-unordered", &allow_unordered, false);
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: the `// non-standard` comment needs to be indented

File: src/rgw/rgw_rest_s3.cc [hunk 2]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
+  if (s->system_request) {
+    s->info.args.get_bool("objs-container", &objs_container, false);
+    const char *shard_id_str = s->info.env->get("HTTP_RGWX_SHARD_ID");
+    if (shard_id_str) {  
+      string err;
+      shard_id = strict_strtol(shard_id_str, 10, &err);
+      if (!err.empty()) {
+        ldout(s->cct, 5) << "bad shard id specified: " << shard_id_str << dendl;
+        return -EINVAL;
+   }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: this should have 6 spaces of indent to match the `if (!err.empty()) {` above

File: src/rgw/rgw_rest_s3.cc [hunk 3]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
+  }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: this should have 4 spaces of indent to match the `if (shard_id_str) {` above

File: src/rgw/rgw_rest_s3.cc [hunk 4]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
+    const char *shard_id_str = s->info.env->get("HTTP_RGWX_SHARD_ID");
+    if (shard_id_str) {  
+      string err;
+      shard_id = strict_strtol(shard_id_str, 10, &err);
+      if (!err.empty()) {
+        ldout(s->cct, 5) << "bad shard id specified: " << shard_id_str << dendl;
+        return -EINVAL;
+   }
+  } 
+   else {
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: the else should be on the same line as the } above, like `} else {`

File: src/rgw/rgw_rest_s3.cc [hunk 5]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
+     shard_id = s->bucket_instance_shard_id;
+    }
+   }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: this indent should be 2 spaces instead of 3

File: src/rgw/rgw_rest_s3.cc [hunk 6]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
+  return 0;
+  }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: this closes the function, so should not have indent

File: src/rgw/rgw_rest_s3.cc [hunk 7]
```diff
@@ -772,69 +772,66 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
-}
-return 0;
+  int ret = get_common_params();
+  if (ret < 0) {
+    return ret;
+  }
+  if (!list_versions) {
+    marker = s->info.args.get("marker");
+  } 
+  else {
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: else goes on the same line as the }, like `} else {`

File: src/rgw/rgw_rest_s3.cc [hunk 8]
```diff
@@ -979,14 +973,11 @@ void RGWListBucket_ObjStore_S3::send_common_response()
...
-
 void RGWListBucket_ObjStore_S3::send_response()
 {
-   
-if(op_ret < 0) {
-    set_req_state_err(s, op_ret);
-  }
+  if (op_ret < 0) {
+      set_req_state_err(s, op_ret);
+    }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: lines 979 and 980 have 2 extra spaces of indent

Comment (albIN7) [reply src/rgw/rgw_rest_s3.cc]: @cbodley This branch is not in my local machine. Updated via github editor.

File: src/rgw/rgw_rest_s3.cc [hunk 9]
```diff
@@ -979,14 +972,11 @@ void RGWListBucket_ObjStore_S3::send_common_response()
...
-
 void RGWListBucket_ObjStore_S3::send_response()
 {
-   
-if(op_ret < 0) {
-    set_req_state_err(s, op_ret);
-  }
+  if (op_ret < 0) {
+        set_req_state_err(s, op_ret);
+      }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: lines 978 and 979 moved in the wrong direction, they now have 4 spaces too many

File: src/rgw/rgw_rest_s3.cc [hunk 10]
```diff
@@ -772,69 +772,65 @@ void RGWGetUsage_ObjStore_S3::send_response()
...
+    const char *shard_id_str = s->info.env->get("HTTP_RGWX_SHARD_ID");
+    if (shard_id_str) {  
+      string err;
+      shard_id = strict_strtol(shard_id_str, 10, &err);
+      if (!err.empty()) {
+        ldout(s->cct, 5) << "bad shard id specified: " << shard_id_str << dendl;
+        return -EINVAL;
+      }
+    } 
+    else {
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: } else {

Comment (albIN7) [reply src/rgw/rgw_rest_s3.cc]: @cbodley updated

**Comment by alimaredia** [comment]
jenkins retest this please

**Comment by alimaredia** [comment]
@albIN7 could you please squash all your commits into one and make sure that commit is signed off.

**Comment by albIN7** [comment]
@alimaredia I am sorry. My laptop crashed last week. This branch is not in my local machine. I updated via github editor.

**Comment by alimaredia** [comment]
@cbodley @albin-antony did these issues get resolved in another PR? If not I'll go ahead and merge this.

**Comment by albin-antony** [comment]
@alimaredia No, the issues are not resolved. I'll raise a new pr if this is not mergeable.

**Comment by alimaredia** [comment]
@albin-antony can you resolve the merge conflicts. Casey and I will review the updated PR and try to get it in ASAP.

**Comment by cbodley** [comment]
hi @albin-antony, i'd like to merge this but there are still conflicts. can you please rebase?

**Comment by cbodley** [comment]
> The base branch does not accept merge commits.

i'm confused why github thinks this is a merge commit

**Comment by ivancich** [comment]
rgw/(non-)verify: http://pulpito.ceph.com/ivancich-2019-09-11_01:23:37-rgw-wip-eric-testing-1-distro-basic-smithi/
rgw/verify: http://pulpito.ceph.com/ivancich-2019-09-13_17:22:14-rgw-wip-eric-testing-1-bisect-6-distro-basic-smithi/

**Comment by ivancich** [comment]
@albin-antony @alimaredia Where is the tracker for this? Will this need backports? Please create the tracker and document any necessary backports, and then I'll complete the merge. Thanks!

**Comment by albin-antony** [comment]
Thanks @ivancich . Created a tracker. We want to backport the cleanup to nautilus.

**Comment by albin-antony** [comment]
jenkins test make check

**Comment by cbodley** [comment]
thanks @albin-antony!

@theanalyst is working on nautilus backports in https://github.com/ceph/ceph/pull/29849 for https://tracker.ceph.com/issues/41408, so we shouldn't need to backport separately

**Comment by ivancich** [comment]
Thanks, @albin-antony . Generally the description of the pr includes a link back to the tracker (part of the cross-linking I mentioned). I've added it to this one. That allows us to go from Tracker to PR and vice versa.

**Pull request merged**